server/schedule/labeler/labeler.go

```diff
@@ -122,9 +178,26 @@ func (l *RegionLabeler) GetLabelRules(ids []string) ([]*LabelRule, error) {
 
 // GetLabelRule returns the Rule with the same ID.
 func (l *RegionLabeler) GetLabelRule(id string) *LabelRule {
-	l.RLock()
-	defer l.RUnlock()
-	return l.labelRules[id]
+	return l.getAndCheckRule(id, time.Now())
+}
+
+func (l *RegionLabeler) getAndCheckRule(id string, now time.Time) *LabelRule {
+	l.Lock()
+	defer l.Unlock()
```

Comment:
can release the lock before save to storage?

Reply:
sorry we can't , since the rule is a pointer get from `labelRules`,  Without lock it could be updated in the other goroutine(s) which is not thread safe.